physics/module_bl_mynn.F90

```diff
@@ -5889,153 +5921,166 @@ SUBROUTINE DMP_mf(                            &
   !   (4) add wspd-dependent limit, when plume model breaks down. (hurricanes)
   !   (5) limit to reduce max plume sizes in weakly forced conditions. This is only
   !       meant to "soften" the activation of the mass-flux scheme.
   ! Criteria (1)
-    NUP2 = max(1,min(NUP,INT(dx*dcut/dl)))
+    maxwidth = min(dx*dcut, lmax)
   !Criteria (2)
-    maxwidth = 1.1*PBLH 
+    maxwidth = min(maxwidth, 1.1*PBLH) 
   ! Criteria (3)
-    maxwidth = MIN(maxwidth,0.5*cloud_base)
+    if ((landsea-1.5) .lt. 0) then  !land
+       maxwidth = MIN(maxwidth, 0.5*cloud_base)
+    else                            !water
+       maxwidth = MIN(maxwidth, 0.9*cloud_base)
+    endif
   ! Criteria (4)
     wspd_pbl=SQRT(MAX(u(kts)**2 + v(kts)**2, 0.01))
     !Note: area fraction (acfac) is modified below
   ! Criteria (5) - only a function of flt (not fltv)
     if ((landsea-1.5).LT.0) then  !land
       !width_flx = MAX(MIN(1000.*(0.6*tanh((flt - 0.050)/0.03) + .5),1000.), 0.)
-      width_flx = MAX(MIN(1000.*(0.6*tanh((flt - 0.040)/0.03) + .5),1000.), 0.) 
+      !width_flx = MAX(MIN(1000.*(0.6*tanh((flt - 0.040)/0.03) + .5),1000.), 0.) 
+      width_flx = MAX(MIN(1000.*(0.6*tanh((fltv - 0.040)/0.04) + .5),1000.), 0.)
     else                          !water
-      width_flx = MAX(MIN(1000.*(0.6*tanh((flt - 0.003)/0.01) + .5),1000.), 0.)
+      !width_flx = MAX(MIN(1000.*(0.6*tanh((flt - 0.003)/0.01) + .5),1000.), 0.)
+      width_flx = MAX(MIN(1000.*(0.6*tanh((fltv - 0.007)/0.02) + .5),1000.), 0.)
+    endif
+    maxwidth = MIN(maxwidth, width_flx)
+    minwidth = lmin
+    !allow min plume size to increase in large flux conditions (eddy diffusivity should be
+    !large enough to handle the representation of small plumes).
+    if (maxwidth .ge. (lmax - 1.0) .and. fltv .gt. 0.2)minwidth = lmin + dlmin*min((fltv-0.2)/0.3, 1.) 
+
+    if (maxwidth .le. minwidth) then ! deactivate MF component
+       nup2 = 0
+       maxwidth = 0.0
     endif
-    maxwidth = MIN(maxwidth,width_flx)
-  ! Convert maxwidth to number of plumes
-    NUP2 = MIN(MAX(INT((maxwidth - MOD(maxwidth,100.))/100), 0), NUP2)
 
-  !Initialize values for 2d output fields:
-  ktop = 0
-  ztop = 0.0
-  maxmf= 0.0
+  ! Initialize values for 2d output fields:
+    ktop = 0
+    ztop = 0.0
+    maxmf= 0.0
 
-  IF ( fltv2 > 0.002 .AND. NUP2 .GE. 1 .AND. superadiabatic) then
-    !PRINT*," Conditions met to run mass-flux scheme",fltv2,pblh
+!Begin plume processing if passes criteria
+if ( fltv2 > 0.002 .AND. (maxwidth > minwidth) .AND. superadiabatic) then
 
     ! Find coef C for number size density N
     cn = 0.
-    d=-1.9  !set d to value suggested by Neggers 2015 (JAMES).
-    !d=-1.9 + .2*tanh((fltv2 - 0.05)/0.15) 
-    do I=1,NUP !NUP2
-       IF(I > NUP2) exit
-       l  = dl*I                            ! diameter of plume
+    d  =-1.9  !set d to value suggested by Neggers 2015 (JAMES).
+    dl = (maxwidth - minwidth)/real(nup-1)
```

Comment:
When calling "real", you should provide the precision, so real(nup-1,kind=kind_phys)

Reply:
fixed